cmd/example-app/main.go

```diff
@@ -273,6 +268,10 @@ func (a *app) handleCallback(w http.ResponseWriter, r *http.Request) {
 	oauth2Config := a.oauth2Config(nil)
 	switch {
 	case code != "":
+		if state := r.FormValue("state"); state != exampleAppState {
```

Comment:
Just wondering if there is a reason why we should not perform this check for "state" in case of 'refresh'